doc/API.md

```diff
@@ -28,17 +32,379 @@ struct {
+}
+```
+'transactions' is a list of 'WalletTransactions' that affect the input address.
+See the documentation for '/wallet/history' for more information.
+
+#### /wallet/seed [GET]
+
+Function: Return the seed that is being used to generate addresses. This seed
+can be used to derive secret keys, and must be kept safe. This call is
+unavailable when the wallet is locked or closed.
+
+Parameters:
+```
+struct {
+	dictionary string
+}
+```
+'dictionary' is the name of the dicitionary that should be used when encoding
+the seed.
+
+Response:
+```
+struct {
+	primarySeed        string
+	addressesRemaining int
+	allSeeds           []string
+}
+```
+'primarySeed' is the seed that is actively being used to generate new addresses
+for the wallet.
+
+'addressesRemaining' is the number of addresses that remain in the primary seed
+until exhaustion has been reached and no more addresses will be generated.
+
+'allSeeds' is a list of all seeds that the wallet references when scanning the
+blockchain for outputs. The wallet is able to spend any output generated by any
+of the seeds, however only the primary seed is being used to generate new
+addresses.
+
+A seed is an encoded version of a 128 bit random seed. The output is 15 words
+chosen from a small dictionary as indicated by the input. The most common
+choice for the dictionary is going to be 'english'. The underlying seed is the
+same no matter what dictionary is used for the encoding. The encoding also
+contains a small checksum of the seed, to help catch simple mistakes when
+copying. The library
+[entropy-mnemonics](https://github.com/NebulousLabs/entropy-mnemonics) is used
+when encoding.
+
+#### /wallet/seed [PUT]
+
+Function: Get a new address from the wallet generated by the primary seed. An
+error will be returned if the seed has been exhausted (there is a limit on the
+number of addresses that can be generated from one seed), or if the wallet is
+locked/closed.
+
+Parameters: none
+
+Response:
+```
+struct {
+	address string
+}
+```
+'address' is the address that cna
+
+#### /wallet/seed [POST]
+
+Function: Fetch a new seed for the wallet. The old seed will be added to the
+list of backup seeds. In general, this call should be avoided unless address
+exhaustion is reached. The encryption password is required to make this change.
+
+Parameters:
+```
+struct {
+	encryptionKey string
+	dictionary    string
+}
+```
+'verification' must be set to true, or an error is returned. This is to prevent
+an implementation mistake where POST is called instead of GET or PUT.
+
+'dictionary' is the name of the dictionary that should be used when encoding
+the newly created seed. 'english' is the most common choice.
+
+Response:
+```
+struct {
+	newSeed string
+}
+```
+'newSeed' is the new seed that will be used as the seed for generating new
+addresses.
+
+#### /wallet/siacoins [PUT]
+
+Function: Send siacoins to an address. The outputs are arbitrarily selected
+from addresses in the wallet.
```

Comment:
PUT is usually used to update the resource referenced by the URL path. So `PUT /wallet/siacoins/${addr}` would make a bit more sense. Even then, PUT (as I understand it) usually means you are overwriting the existing value, whereas here we are adding to it.